### PR TITLE
[Testing] Add A3/H100 to the GPU nvml/dcgm tests

### PR DESCRIPTION
**Pull request by LujieDuan**
## Description
Add A3 (with H100 80GB GPUs) to the NVML and DCGM integration tests. 

Due to quota constraints, the A3 tests only run in nightlies (on `ubuntu-os-cloud:ubuntu-minimal-2004-lts`). 

## Related issue
[b/296048457](http://b/296048457)

## How has this been tested?
Temporarily changed the presubmit to run on `ubuntu-os-cloud:ubuntu-minimal-2004-lts` and the tests were passing. 

## Checklist:
- Unit tests
  - [x] Unit tests do not apply.
  - [ ] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [x] This PR introduces no user visible changes.
  - [ ] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [x] This PR introduces no new features.
  - [ ] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: integration_test/third_party_apps_test/main_test.go [hunk 1]
```diff
@@ -819,40 +820,53 @@ var defaultApps = map[string]bool{
...
+		machineType:      "g2-standard-4",
+		availableZone:    "us-central1-a",
+		acceleratorCount: 1,
+	},
+	"h100": {
+		model:            "h100",
+		fullName:         "nvidia-h100-80gb",
+		machineType:      "a3-highgpu-8g",
+		availableZone:    "us-west1-a",
+		acceleratorCount: 8,
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_test/main_test.go:869]: Can we parse this out of the machine type (`-8g`)?

Comment (LujieDuan) [reply integration_test/third_party_apps_test/main_test.go:869]: Yes I think we can! The naming seems consistent with H100(a3) and A100(a2). For a2, there are "a2-ultragpu-1g", "a2-ultragpu-2g", "a2-ultragpu-4g" and "a2-ultragpu-8g" and for a3 "a3-highgpu-8g". Other machine types can take the default value of "--accelerator=count=1" when the machine type is not ended with "-\dg". Will try this in this PR!

**Comment by LujieDuan** [comment]
TODO: Because of the H100 quota, need to make the test to skip presubmit and only run in nightlies.

File: integration_test/third_party_apps_test/applications/dcgm/metadata.yaml
Comment (igorpeshansky) [inline integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:1]: Also, your PR description says "nvml test", but this is changing the `dcgm` test instead… Which is correct? Or did you mean to change both?

Comment (LujieDuan) [reply integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:1]: Ideally H100 should be tested for both `nvml` and `dcgm` but the current quota only allows us to run one at a time. `nvml` passed in [a previous commit](https://github.com/GoogleCloudPlatform/ops-agent/pull/1725/commits/414f43be0f5139d776c253a426faf459b64be282) and then I switched the test to verify `dcgm`. Will test both when quota becomes available. Sorry for the confusion!

File: integration_test/third_party_apps_test/main_test.go (hunk 1 above)
Comment (igorpeshansky) [reply integration_test/third_party_apps_test/main_test.go:887]: Is this still something we'd want to do?

File: integration_test/third_party_apps_test/applications/dcgm/metadata.yaml
Comment (LujieDuan) [reply integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:1]: Enabled for both DCGM and NVML.

File: integration_test/third_party_apps_test/main_test.go (hunk 1 above)
Comment (LujieDuan) [reply integration_test/third_party_apps_test/main_test.go]: Done!

File: integration_test/third_party_apps_test/applications/dcgm/metadata.yaml [hunk 2]
```diff
@@ -44,7 +44,7 @@ gpu_platforms: # p4, p100 don't emit DCGM profiling metrics
       - ubuntu-os-cloud:ubuntu-2204-lts
   - model: h100
     platforms:
-      - debian-cloud:debian-11
+      - ubuntu-os-cloud:ubuntu-minimal-2004-lts # due to H100 quota, choose a image from the exhaustive list to skip presubmits
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:47]: Nit: `an image` or `one image`.

File: integration_test/third_party_apps_test/applications/nvml/metadata.yaml [hunk 3]
```diff
@@ -49,6 +49,9 @@ gpu_platforms:
       - suse-cloud:sles-15
       - ubuntu-os-cloud:ubuntu-2004-lts
       - ubuntu-os-cloud:ubuntu-2204-lts
+  - model: h100
+    platforms:
+      - ubuntu-os-cloud:ubuntu-minimal-2004-lts # due to H100 quota, choose a image from the exhaustive list to skip presubmits
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_test/applications/nvml/metadata.yaml:54]: Nit: `an image` or `one image`.

File: integration_test/third_party_apps_test/main_test.go [hunk 4]
```diff
@@ -929,6 +921,19 @@ func incompatibleOperatingSystem(testCase test) string {
 	return "" // We are testing on a supported image for this app.
 }
 
+// getAcceleratorCount parses the machine type string and get the accelerator
+// count for that machine type. For machine types that support multiple GPU
+// configurations (n1-standard-2 can have 1, 2, 4 or 8 V100 attached), use 1
+// as the default accelerator count
+// https://cloud.google.com/compute/docs/gpus
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_test/main_test.go:928]: A couple of nits: `gets` or even `extracts`, and this comment is missing a trailing `.`. I would also put the doc link inline next to "machine type". For example:
```suggestion
// getAcceleratorCount extracts the accelerator count from the machine type
// string (https://cloud.google.com/compute/docs/gpus). For machine types that
// support multiple GPU configurations (n1-standard-2 can have 1, 2, 4 or 8
// V100 attached), use 1 as the default accelerator count.
```

**Review by igorpeshansky — COMMENTED** [review]
Minor nits.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Pull request merged**